docs/Contribute_To_Frontend_Quick_Guide.md

```diff
@@ -0,0 +1,243 @@
+
+
+If yarn is installed properly, Install the external library dependencies
+
+Type the command:
+```
+yarn install
+```
+Build the code and Pass all required tests listed in **pacakage.json**
+```
+yarn watch
+```
+
+The watch command will leave the code in a state that will "watch" for further edits. If desired, the "watch" may be stopped by hitting ctrl-c
+
+### In a seperate terminal, log into the **adm** node
+* vagrant ssh adm
+
+Log in as root
+* su -
+
+Go to the cloned **GUI** repo on the /vagrant mount point where the code edits were made.
+* cd /vagrant/GUI
+
+* **yarn link**
+
+```
+yarn link v0.27.5
+success Registered "@iml/gui".
+info You can now run `yarn link "@iml/gui"` in the projects where you want to use this module and it will be used instead.
+Done in 0.60s.
+
+```
+The "yarn link" provides the ability to replace the 'installed' GUI that came with the installed IML with 'this' GUI with code edits.
+
+### Stop the IML running services
+* chroma-config stop
+
+### Replace the **gui** that is running on the **adm** node with the newly cloned **gui**
+
+The gui modules currently reside at /usr/share/chroma-manager/ui-modules on the running system.
+
+* cd /usr/share/chroma-manager/ui-modules
+* yarn link "@iml/gui"
+
+### Start the IML services
+* chroma-config start
+
+## In a browser, go to the IML location
+* [https://ct7-adm.lfs.local:8443](https://ct7-adm.lfs.local:8443)
+
+## Verify that the small change worked.
+
+It is possible that the browser cache may require refreshing for the change to take place.
+
+* Click on 
+    * Configuration 
+    * Verify that the pull down menu has the item: "My Servers"
+    * The image below also shows a change for "My Power Control"
+
+![iml_flow](md_Graphics/config_my_servers.png)
+
+---
+# Congratulations! You just made a change to the IML Frontend code.
+---
+
+## The process outlined above is the basic technique for modifying the Frontend IML code.
+
+## A note about starting and stopping chroma-config
+* The amount of time to complete 
+    * chroma-config start
+    * chroma-config stop
+
+* can be time consuming, however, the commands are still an effective way to ensure that all processes start and stop.
+
+* A faster method to restart just the "view_server" process can be accomplished as follows:
+
+    * **supervisorctl -c /usr/share/chroma-manager/production_supervisord.conf restart view_server**
+
+
+* To check the IML services status
+
+   * **supervisorctl -c /usr/share/chroma-manager/production_supervisord.conf status**
+
+
+# Pushing the code change to github
+
+### On your local machine, i.e., not the vagrant VM:
+
+* cd ~/vagrant-projects/vhpc/GUI
+
+Ensure you are on the proper branch
+* git branch
+
+Otherwise, change to the **my-fix** branch
+* git checkout my-fix
+
+Add and Commit the change.
+
+* git status
+
+* git add source/iml/app/app-states.js
+
+* git commit -s 
+
+This will cause the **eslint**, **flow** and **unit** tests to run.
```

Comment:
Might want to change this to say "cause the pre-commit hooks to run"

Reply:
changed